core/src/loader.rs

```diff
@@ -122,6 +137,30 @@ pub fn load_associated_token_account(
     Ok(())
 }
 
+/// Loads the account as a token account, returning an error if it is not.
+///
+/// # Arguments
+/// * `token_account` - The account to load the token account from
+///
+/// # Returns
+/// * `Result<(), ProgramError>` - The result of the operation
+pub fn load_token_account(
+    token_account: &AccountInfo,
+    token_program_info: &AccountInfo,
+) -> Result<(), ProgramError> {
+    if token_account.owner.ne(token_program_info.key) {
+        msg!("Account is not owned by the token program");
+        return Err(ProgramError::InvalidAccountOwner);
+    }
+
+    if token_account.data_is_empty() {
+        msg!("Account data is empty");
+        return Err(ProgramError::InvalidAccountData);
+    }
+
```

Comment:
should deser the account to make sure its not a token mint/something else

Reply:
I added deserializing token account like `load_token_mint`.

```rust
    let data = token_account.try_borrow_data()?;
    let _ = StateWithExtensions::<spl_token_2022::state::Account>::unpack(&data)?;

```